revise /customnode/installed api

ltdrdata opened the pull request:
improved: don't fetch data from cnr for the api
improved: change format `{<cnr id>: <version>} -> {<module>: {'ver': <version>, 'cnr_id': <cnr id>, 'enabled': <enable/disable status>}}`
improved: add `mode=imported` for startup snapshot
* `/customnode/installed` - current snapshot
* `/customnode/installed?mode=imported` - startup snapshot

example:
```
{
      "ComfyUI-GGUF": { // unknown
        "ver": "3dc384b23366983b28222cbf681b808053949a43",
        "cnr_id": "",
        "enabled": true
      }, 
      "comfyui-impact-pack": {  // nightly
        "ver": "12e838a3209202f4fbe06f1af8607cb1417560f5",
        "cnr_id": "comfyui-impact-pack",
        "enabled": true
      },
      "comfyui_layerstyle": {  // cnr
        "ver": "1.0.90",
        "cnr_id": "comfyui_layerstyle",
        "enabled": true
      },
      "comfyui-supir@nightly": { // disabled
        "ver": "53fc4f82f139e0875e1f4f3716fbeafa073e4242",
        "cnr_id": "comfyui-supir",
        "enabled": false
      },
}

```

ltdrdata committed:
revise /customnode/installed

improved: don't fetch data from cnr for the api
improved: change format {<cnr id>: <version>} -> {<module>: [<version>, <cnr id>]}

ltdrdata committed:
fix condition

ltdrdata committed:
improved: add `mode=imported` for startup snapshot

`/customnode/installed` - current snapshot
`/customnode/installed?mode=imported` - startup snapshot

ltdrdata committed:
improved: move cache dir to user directory

ltdrdata committed:
Merge branch 'main' into improve/api-installed

ltdrdata committed:
improved: move cache dir to user directory

ltdrdata committed:
Merge branch 'main' into improve/api-installed

ltdrdata committed:
Merge branch 'main' into improve/api-installed

huchenlei commented on glob/manager_core.py:
```diff
@@ -1379,6 +1381,48 @@ async def migrate_unmanaged_nodes(self):
+
+    for x in get_custom_nodes_paths():
+        for y in os.listdir(x):
+            if y == '__pycache__' or y.endswith('.disabled'):
```
I think the installed node packs should contain installed but disabled packs.

ltdrdata replied:
Oh.. ok.
If a full list is required, I will modify it to also provide the disable status.

ltdrdata committed:
modified: /customnodes/installed
- show whole nodes including disabled
- format changed `key -> list` to `key -> dict`

ltdrdata committed:
Merge branch 'main' into improve/api-installed

ltdrdata committed:
fixed: doesn't show disabled node pack properly.

huchenlei committed:
Update workflow-metadata.js

jmgomez commented:
FYI this PR caused a lot of nodes to crash the workflow (i.e. when the node is placed in the wokflow, you cant run the batch anymore as the `installedNodes` is not correctly filled in. It also makes obsolete this tutorial: https://docs.comfy.org/custom-nodes/custom_node_walkthrough

Edit: I see this is now fixed
